src/main/java/duke/Command.java

```diff
@@ -0,0 +1,219 @@
+package duke;
+
+import java.time.LocalDate;
+import java.util.Objects;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.TaskType;
+import duke.task.Todo;
+
+/**
+ * Encapsulates a single command for the Duke program.
+ */
+public class Command {
+    protected TaskType taskType;
+    protected Integer index;
+    protected String description;
+    protected LocalDate date;
+
+    /**
+     * Creates an invalid command.
+     */
+    public Command() {
+        taskType = null;
+    }
+
+    /**
+     * Creates a Command representing a task of either LIST/BYE.
+     * @param taskType the type of task
+     */
+    public Command(TaskType taskType) {
+        switch (taskType) {
+        case LIST:
+            // Fallthrough
+        case BYE:
+            break;
+        default:
+            throw new IllegalArgumentException("You need to provide an index or description.");
+        }
+        this.taskType = taskType;
+        index = null;
+        description = null;
+        date = null;
+    }
+
+    /**
+     * Creates a Command representing a task of either DELETE or DONE.
+     * @param taskType the type of task
+     * @param index index of TaskList the action is to be applied to
+     */
+    public Command(TaskType taskType, Integer index) {
+        switch (taskType) {
+        case DELETE:
+            // Fallthrough
+        case DONE:
+            break;
+        default:
+            throw new IllegalArgumentException("A description parameter is required for this kind of TaskType");
+        }
+        this.taskType = taskType;
+        this.index = index;
+        description = null;
+        date = null;
+    }
+
+    /**
+     * Creates a Command representing a task of TODO.
+     * @param taskType the type of task
+     * @param description the description of the task
+     */
+    public Command(TaskType taskType, String description) {
+        switch (taskType) {
+        case TODO:
```

Comment:
Minor nitpick but could have included a comment `// Fallthrough` like you did in other parts of your code to make it clearer to the reader

Reply:
Thanks for catching that. I'll fix this in the next commit too.